internal/rpcnetworkclient/client.go

```diff
@@ -0,0 +1,219 @@
+package rpcnetworkclient
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/stellar/friendbot/internal"
+	"github.com/stellar/go/amount"
+	rpcclient "github.com/stellar/go/clients/rpcclient"
+	protocol "github.com/stellar/go/protocols/rpc"
+	"github.com/stellar/go/xdr"
+)
+
+const (
+	submitTransactionTimeout = 30 * time.Second
+	backoffInitialInterval   = 500 * time.Millisecond
+	backoffMaxInterval       = 3500 * time.Millisecond
+	statusError              = "ERROR"
+)
+
+// NetworkError wraps an RPC error and implements the internal.NetworkError interface.
+type NetworkError struct {
+	err                 error
+	notFound            bool
+	timeout             bool
+	resultXDR           string
+	diagnosticEventsXDR []string
+}
+
+// IsNotFound returns true if the error indicates the requested resource was not found.
+func (e *NetworkError) IsNotFound() bool {
+	return e.notFound
+}
+
+// IsBadSequence returns true if the error indicates a bad sequence number.
+func (e *NetworkError) IsBadSequence() bool {
+	result, err := e.ResultXDR()
+	if err != nil {
+		return false
+	}
+	return result.Result.Code == xdr.TransactionResultCodeTxBadSeq
+}
+
+// ResultXDR parses and returns the transaction result XDR.
+func (e *NetworkError) ResultXDR() (xdr.TransactionResult, error) {
+	var result xdr.TransactionResult
+	resultXDR, err := e.ResultString()
+	if err != nil {
+		return result, err
+	}
+	if err := xdr.SafeUnmarshalBase64(resultXDR, &result); err != nil {
+		return result, err
+	}
+	return result, nil
+}
+
+// IsTimeout returns true if the error indicates a timeout occurred.
+func (e *NetworkError) IsTimeout() bool {
+	return e.timeout
+}
+
+// ResultString returns the result string from the error, if available.
+func (e *NetworkError) ResultString() (string, error) {
+	if e.resultXDR == "" {
+		return "", fmt.Errorf("no result XDR available")
+	}
+	return e.resultXDR, nil
+}
+
+// DiagnosticEventStrings returns the diagnostic event XDR strings, if available.
+func (e *NetworkError) DiagnosticEventStrings() []string {
+	return e.diagnosticEventsXDR
+}
+
+// Error implements the error interface.
+func (e *NetworkError) Error() string {
+	return e.err.Error()
+}
+
+// Unwrap returns the underlying error.
+func (e *NetworkError) Unwrap() error {
+	return e.err
+}
+
+// NetworkClient wraps an RPC client and implements the internal.NetworkClient interface.
```

Comment:
could be good to validate that it matches the interface like we do in rpc/hz, e.g.:

```go
var _ internal.NetworkClient = (*NetworkClient)(nil)
```

Go otherwise doesn't make it obvious (without a comment) that a type fulfills an interface

Reply:
Great idea. b9adf0e4e89da615191277980d89a4fe8bf758e7